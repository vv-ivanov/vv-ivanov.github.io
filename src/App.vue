<template>
  <div id="app">
    <Link>
      <a class="active" href="/main">Main</a>
      <a href="/chat">Chat</a>
      <a href="/spa">Spa</a>
    </Link>
    <router-view></router-view>
  </div>
</template>

<script>
import Link from "@/components/Link";
export default {
  name: "App",
  components: {
    Link,
  },
  computed: {},
  mounted() {},
  destroyed() {},
  methods: {},
};
</script>
<style lang="scss">
html,
body {
  font-family: "Roboto", sans-serif;
  min-height: 100%;
  height: 100%;
  margin: 0;
  // #app {
  //   min-height: 100%;
  //   // overflow-x: hidden;
  // }
}
* {
  box-sizing: border-box;
}

@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap");
</style>
