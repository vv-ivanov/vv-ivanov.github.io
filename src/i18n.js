import Vue from "vue";
import VueI18n from "vue-i18n";

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context(
    "./locales",
    true,
    /[A-Za-z0-9-_,\s]+\.json$/i
  );
  const messages = {};
  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });
  return messages;
}

function parseDomainForLocale() {
  let hostName = window.location.hostname;
  let domain = hostName.substring(
    hostName.lastIndexOf(".", hostName.lastIndexOf(".") - 1) + 1
  );
  if (domain == "sutochno.com") {
    return "en";
  } else {
    return "ru";
  }
}

export default new VueI18n({
  pluralizationRules: {
    /** Key - language to use the rule for, 'ru', in this case */
    /** Value - function
     * @param choice {number} a choice index given by the input to $tc: `$tc('path.to.rule', choiceIndex)`
     * @param choicesLength {number} an overall amount of available choices
     * @returns a final choice index to select plural word by
     **/
    ru: function (choice, choicesLength) {
      if (choice === 0) {
        return 0;
      }
      // pl: n => n < 2 ? n : n % 10 >= 2 && n % 10 <= 4 && (n % 100 < 10 || n % 100 >= 20) ? 2 : 3

      const teen = choice > 10 && choice < 20;
      const endsWithOne = choice % 10 === 1;

      if (!teen && endsWithOne) {
        return 1;
      }
      if (
        !teen &&
        choice % 10 >= 2 &&
        choice % 10 <= 4 &&
        (choice % 100 < 10 || choice % 100 >= 20)
      ) {
        return 2;
      }
      return choicesLength < 4 ? 2 : 3;
    },
  },
  locale: parseDomainForLocale() || process.env.VUE_APP_I18N_LOCALE || "ru",
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || "ru",
  messages: loadLocaleMessages(),
});
