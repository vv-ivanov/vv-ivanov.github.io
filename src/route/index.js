import Vue from "vue";
import Router from "vue-router";

Vue.use(Router);

export default new Router({
  base:
    process.env.NODE_ENV === "production" ? process.env.VUE_APP_BASEPATH : "/",
  routes: [
    {
      path: "/main",
      name: "Main",
      component: () => import("@/components/pages/Main.vue"),
    },
    {
      path: "/chat",
      name: "Chat",
      component: () => import("@/components/pages/Chat.vue"),
    },
    {
      path: "/spa",
      name: "Spa",
      component: () => import("@/components/pages/Spa.vue"),
    },
  ],
  mode: "history",
});
