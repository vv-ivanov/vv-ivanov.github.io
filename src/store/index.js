import Vue from "vue";
import Vuex from "vuex";
import schedule from "./modules/schedule";
import VueLocalStorage from "vue-localstorage";

Vue.use(Vuex);
Vue.use(VueLocalStorage);

const debug = process.env.NODE_ENV !== "production";

export default new Vuex.Store({
  modules: {
    schedule,
  },
  strict: debug,
});
