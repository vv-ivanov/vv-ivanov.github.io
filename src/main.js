// The Vue build version to load with the `import` command
// (runtime-only or standalone) has been set in webpack.base.conf with an alias.
import "core-js/stable";
import Vue from "vue";
import router from "./route/index";

import App from "./App.vue";
// other imports removed for brevity

// common components
import store from "./store/";
import i18n from "./i18n";

const isProd = process.env.NODE_ENV === "production";
const isModernBuild = process.env.VUE_APP_MODERN_BUILD === true;

//other
Vue.config.productionTip = false;

//init app
new Vue({
  router,
  store,
  i18n,
  render: (h) => h(App),
}).$mount("#app");
