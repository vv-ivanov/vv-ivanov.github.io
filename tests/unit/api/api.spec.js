import { shallowMount } from "@vue/test-utils";
import { Users } from "../../../src/api/";

describe("API: Users", () => {
  it("getUserData", async () => {
    const RESPONSE = await Users.getUserData();
    expect(RESPONSE).toMatchSnapshot();
  });
});
