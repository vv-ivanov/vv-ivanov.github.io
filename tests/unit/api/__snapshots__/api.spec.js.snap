// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API: Users getUserData 1`] = `
Object {
  "data": Object {
    "actions": Array [],
    "data": Object {
      "balance": "37 907 <span class=\\"rubl\\">&#8381;</span>",
      "cnt_booking_owner": 0,
      "complete_publication": 75,
      "complete_publication_url": "/cabinet/objects/edit/648300/step4.html",
      "estimated": Array [],
      "extension": Object {
        "menu": Array [
          Object {
            "name": "showOwnerInfo",
            "title": "Информация о владельце",
            "trigger": true,
          },
          Object {
            "name": "showExtendedFilters",
            "title": "Дополнительные фильтры",
            "trigger": true,
          },
        ],
        "rev": "15746",
        "src": "b_panel",
      },
      "hint_set_mode": false,
      "hot_cnt_to_me": 0,
      "hot_cnt_to_me_text": "0 запросов на подбор жилья",
      "hot_offers_cnt_my": 0,
      "hot_offers_cnt_to_me": 0,
      "hot_orders": Array [
        Object {
          "cnt": "0",
          "dateBegin": "02.08",
          "dateEnd": "03.08",
          "id": "2386434",
          "location": "Санкт-Петербург",
        },
      ],
      "hotels_count": 2,
      "invoices": Object {
        "cnt": 12,
        "list": Array [
          Object {
            "amount_format": "300 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "5 августа 2019",
            "id": "12616530",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
          Object {
            "amount_format": "2 625 <span class=\\"rubl\\">&#8381;</span>",
            "date_add_format": "22 июля 2019",
            "id": "12616399",
          },
        ],
      },
      "is_nd": 0,
      "messages": Object {
        "exists": true,
        "modules": Object {
          "messages": Object {
            "list": Array [
              Object {
                "date_add": "1 июня 2019",
                "url": "/cabinet/notices/message/47426384",
                "user_name": "Суточно.ру",
                "user_photo": "//static.sutochno.test/doc/files/users/0/0/2/small/57d4f585bbebd.jpg",
              },
              Object {
                "date_add": "1 июня 2019",
                "url": "/cabinet/notices/message/47412970",
                "user_name": "Суточно.ру",
                "user_photo": "//static.sutochno.test/doc/files/users/0/0/2/small/57d4f585bbebd.jpg",
              },
              Object {
                "date_add": "1 июня 2019",
                "url": "/cabinet/notices/message/47410528",
                "user_name": "Суточно.ру",
                "user_photo": "//static.sutochno.test/doc/files/users/0/0/2/small/57d4f585bbebd.jpg",
              },
              Object {
                "date_add": "29 мая 2019",
                "url": "/cabinet/notices/message/47300452",
                "user_name": "Суточно.ру",
                "user_photo": "//static.sutochno.test/doc/files/users/0/0/2/small/57d4f585bbebd.jpg",
              },
              Object {
                "date_add": "28 мая 2019",
                "url": "/cabinet/notices/message/47266188",
                "user_name": "Суточно.ру",
                "user_photo": "//static.sutochno.test/doc/files/users/0/0/2/small/57d4f585bbebd.jpg",
              },
            ],
            "new": 440,
          },
          "objects": Object {
            "has_reviews": false,
          },
          "orders": Object {
            "list": Array [
              Object {
                "date_add": "26 минут назад",
                "url": "/cabinet/notices/8940798",
                "user_name": "MAG Android",
                "user_photo": "//static.sutochno.test/doc/files/users/4/292/155/small/5ce78c9680994.jpg",
              },
              Object {
                "date_add": "36 минут назад",
                "url": "/cabinet/notices/8940794",
                "user_name": "Яна",
                "user_photo": "//static.sutochno.test/doc/i/default-avatar_small.jpg",
              },
              Object {
                "date_add": "14 августа 2019",
                "url": "/cabinet/notices/8940786",
                "user_name": "Elmira",
                "user_photo": "//static.sutochno.test/doc/i/default-avatar_small.jpg",
              },
              Object {
                "date_add": "1 августа 2019",
                "url": "/cabinet/notices/8940604",
                "user_name": "Яна",
                "user_photo": "//static.sutochno.test/doc/i/default-avatar_small.jpg",
              },
              Object {
                "date_add": "12 августа 2019",
                "url": "/cabinet/notices/8940702",
                "user_name": "Виктор",
                "user_photo": "//static.sutochno.test/doc/files/users/0/848/400/small/5d3ae52889912.png",
              },
            ],
            "new": 22,
          },
        },
        "new": 462,
      },
      "new_orders": 0,
      "objects": Object {
        "favorites": 0,
        "favorites_cities_groups": Array [],
        "favorites_ids": Array [],
        "text": "Нет объектов",
      },
      "objects_count": 3,
      "pending_reviews_bonus": Array [],
      "searchAppEnabled": "1",
      "uid": null,
      "user": Object {
        "avatar": "//static.sutochno.test/doc/files/users/0/7/52/small/5d53f15170cfe.jpeg",
        "id": "7052",
        "name": "Иван",
      },
      "user_is_client": 153,
      "user_is_client_loyal": 2,
      "user_is_moder": true,
      "user_is_newbie": false,
      "user_is_owner": 20,
      "user_is_superhost": false,
    },
    "errors": Array [],
    "platform": "js",
    "success": true,
    "timestamp": 1565870612,
    "uuid": "1536304442",
  },
}
`;
